core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -215,6 +215,12 @@ protected function prepare($objPage)
 		// Meta robots tag
 		$this->Template->robots = $responseContext->get(HtmlHeadBag::class)->getMetaRobots();
 
+		// Do not search the page if the query has a key that is in TL_NOINDEX_KEYS
+		if (preg_grep('/^(' . implode('|', $GLOBALS['TL_NOINDEX_KEYS']) . ')$/', array_keys($_GET)))
```

Comment:
I would like to get rid of this too. Makes no sense. It seems we're using this to somehow prevent indexing for filtering (pagination, calendar range filters etc.) rather than correctly setting the right canonical link which would have been the right solution ever since. But that's for another PR thus that remains here for now.

Reply:
I also don't understand why indexing a pagination (e.g. product list) wouldn't make sense... 🤷‍♂️

Reply:
I suggest to create a separate issue to discuss the `TL_NOINDEX_KEYS`?

Reply:
Done https://github.com/contao/contao/issues/3097